ui/hooks/useGasFeeInputs.js

```diff
@@ -164,6 +180,40 @@ export function useGasFeeInputs(defaultEstimateToUse = 'medium') {
     estimatedGasFeeTimeBounds,
   } = useGasFeeEstimates();
 
+  // This hook keeps track of a few pieces of transitional state. It is
+  // transitional because it is only used to modify a transaction in the
+  // metamask (background) state tree.
+  const [maxFeePerGas, setMaxFeePerGas] = useState(
+    transaction?.txParams?.maxFeePerGas
+      ? Number(hexWEIToDecGWEI(transaction.txParams.maxFeePerGas))
+      : null,
+  );
+  const [maxPriorityFeePerGas, setMaxPriorityFeePerGas] = useState(
+    transaction?.txParams?.maxPriorityFeePerGas
+      ? Number(hexWEIToDecGWEI(transaction.txParams.maxPriorityFeePerGas))
+      : null,
+  );
+  const [gasPrice, setGasPrice] = useState(
+    transaction?.txParams?.gasPrice
+      ? Number(hexWEIToDecGWEI(transaction.txParams.gasPrice))
+      : null,
+  );
```

Comment:
This change is needed below:
![Screenshot from 2021-07-27 04-59-04](https://user-images.githubusercontent.com/7499938/127113947-2a28a523-dac8-4991-b62c-13c45656e906.png)

Reply:
Because the property on `txParams` is called `gas` and no `gasLimit`, and because gas/gasLimit is not in WEI or GWEI (it does not have a unit)

Reply:
> the property on txParams is called `gas` and not `gasLimit`

We do need to better improve this naming situation in the codebase. Ideally, it would be the same everywhere... but that is for a different PR